consensus/polybft/consensus_runtime.go

```diff
@@ -855,94 +845,237 @@ func (c *consensusRuntime) getCommitmentToRegister(epoch *epochMetadata,
 	}, nil
 }
 
-func validateVote(vote *MessageSignature, epoch *epochMetadata) error {
-	// get senders address
-	senderAddress := types.StringToAddress(string(vote.From))
-	if !epoch.Validators.ContainsAddress(senderAddress) {
-		return fmt.Errorf(
-			"message is received from sender %s, which is not in current validator set",
-			vote.From,
-		)
+func (c *consensusRuntime) IsValidBlock(proposal []byte) bool {
+	if err := c.fsm.Validate(proposal); err != nil {
+		c.logger.Error("failed validate proposal", "error", err)
+
+		return false
 	}
 
-	return nil
+	return true
+}
+
+func (c *consensusRuntime) IsValidSender(msg *proto.Message) bool {
+	err := c.fsm.ValidateSender(msg)
```

Comment:
What do you think about moving that `ValidateSender` logic to this `IsValidSender`? As far as I see, you only need `validators` field from `fsm` to validate sender. To me, it feels more natural for that logic to be in `runtime` (`fsm` is more about validating and building proposal).

Reply:
The idea is that runtime is only an adapter to real backend, which is FSM. So, any kind of validation and block manipulation is done on fsm. That is why we have that method on fsm.